packages/@aws-cdk/aws-eks/lib/cluster-resource.ts

```diff
@@ -117,7 +147,21 @@ export class ClusterResource extends Construct {
       resourceType: CLUSTER_RESOURCE_TYPE,
       serviceToken: provider.serviceToken,
       properties: {
-        Config: props,
```

Comment:
The whole idea was that `props` can be passed as is to the resource. Why do we need to cherry pick each prop?

Reply:
This pass-through creates a coupling between the structure of `ClusterResourceProps` and the structure of `CreateClusterRequest`. Until now, they were indeed the same (somewhat incidentally I think). The introduction of these new properties now creates a deviation because `ClusterResourceProps.resourcesVpcConfig` cannot contain them, while `CreateClusterRequest.resourcesVpcConfig` does have them. 

This resulted in something like this:

```typescript
const config = {
  ...props,
  resourcesVpcConfig: {
    ...props.resourcesVpcConfig, // for subnets and security groups
    endpointPublicAccess: props.endpointAccessConfig.endpointPublicAccess,
    endpointPrivateAccess: props.endpointAccessConfig.endpointPrivateAccess,
    publicAccessCidrs: props.endpointAccessConfig.publicAccessCidrs,
  }
}

// we have to remove this otherwise it causes a validation error in the SDK as an unexpected property
// in the request.
delete config.endpointAccessConfig
```

Which IMO is messy, and has the potential to grow even messier as we add new features that aren't supported by CFN. So I opted to be fully explicit, and not rely on those structures being the same.

WDYT? I personally prefer being explicit and would even type it as `aws.EKS.CreateClusterRequest` if that interface was available in this context.  

Regardless, I removed the `props.endpointAccessConfig` nesting and just hoisted these properties to `props`.

Reply:
Yes, that's pretty messy. I am wondering if there is any value in trying to shove the endpoint access config into `Config`... Let's just pass it as a separate set of properties alongside `Config`. This will allow us to model them without any booleans that don't pass through well.

Something like this:

```ts
export interface ClusterResourceProps {
  readonly cfnProps: CfnClusterProps;
  readonly access?: 'public' | 'private' | 'public-private';
  readonly accessCidrs: string[];
}

// and then
new CustomResource(this, 'Resource', {
  properties: {
    Config: props.cfnProps,
    Access: props.access,
    AccessCidrs: props.accessCirds,
  }
});
```

Something like that maybe?

I also have the logging configuration in mind which suffers from a similar issue with boolean properties, see the abandoned https://github.com/aws/aws-cdk/pull/8497), so I think just adding additional properties to the custom resource that designed for that use case might be a better approach

Reply:
I'm not sure I see the benefit in this. Is it just so that we don't have to cherry-pick CFN properties that happen to match `CreateClusterRequest` properties?

These properties are only used internally for `ClusterResource` instantiation, meaning that any property we want to add will still have to be explicitly exposed in `eks.ClusterProps`. Having another layer to expose it to doesn't seem like much work. 

Its also somewhat confusing, isn't `Access` and `AccessCidrs` also a form of `Config`? This separation doesn't feel natural, and I don't see a reason we can't have:

```typescript
Config: {
  ...,
  logging: {
  ...
  }
}
```

> I am wondering if there is any value in trying to shove the endpoint access config

We don't have to, but they do logically belong in the VPC config, and since i think explicitly creating the structure is better, i opted to put in there. 

Since this doesn't have public API implications, i'm going to leave it as is, let me know if you have strong objections to it.